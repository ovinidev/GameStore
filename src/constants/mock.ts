import { Game } from "@interfaces/games";

export const GAME_MOCK: Game = {
	title: "Game Beta",
	description: "A thrilling game",
	genre: "Action",
	id: 1,
	image: "image-url",
	platform: "PC",
	price: 60,
	release_date: "2024-08-01",
};

export const GAMES_MOCK: Game[] = [
	{
		title: "Game Beta",
		description: "A thrilling game",
		genre: "Action",
		id: 1,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
	{
		title: "Game Alpha",
		description: "A thrilling game",
		genre: "Action",
		id: 2,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
];

export const CART_MOCK: Game[] = [
	{
		title: "Game Beta",
		description: "A thrilling game",
		genre: "Action",
		id: 1,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
	{
		title: "Game Alpha",
		description: "A thrilling game",
		genre: "Action",
		id: 2,
		image: "image-url",
		platform: "PC",
		price: 60,
		release_date: "2024-08-01",
	},
];
