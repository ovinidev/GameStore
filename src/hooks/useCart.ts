import { useAddCart, useDeleteCart } from "@mutations/cart";
import { useGetCart } from "@queries/cart";

export const useCart = () => {
	const { data } = useGetCart();
	const { mutateAsync: addToCart } = useAddCart();
	const { mutateAsync: deleteFromCart } = useDeleteCart();

	return {
		addToCart,
		deleteFromCart,
		data,
	};
};
