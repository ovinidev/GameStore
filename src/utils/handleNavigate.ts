import { Router } from "expo-router";

interface HandleNavigateProps {
	router: Router;
	id: number;
}

export const handleNavigate = ({ router, id }: HandleNavigateProps) => {
	router.push(`/${id}`);
	router.setParams({
		id,
	});
};
