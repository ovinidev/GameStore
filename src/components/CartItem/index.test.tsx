import { fireEvent, render } from "@testing-library/react-native";
import { CartItem } from "./index";
import { GAME_MOCK } from "@constants/mock";
import { handleNavigate } from "@utils/handleNavigate";
import { useCart } from "@hooks/useCart";

jest.mock("@hooks/useCart", () => ({
	useCart: jest.fn().mockReturnValue({
		deleteFromCart: jest.fn(),
	}),
}));

jest.mock("@utils/handleNavigate", () => ({
	handleNavigate: jest.fn(),
}));

jest.mock("expo-router", () => ({
	useRouter: jest.fn().mockReturnValue({
		push: jest.fn(),
		setParams: jest.fn(),
	}),
}));

jest.mock("@expo/vector-icons/Feather", () => "Feather");

describe("CartItem", () => {
	it("should be able to match the snapshot", () => {
		const { toJSON } = render(<CartItem data={GAME_MOCK} />);

		expect(toJSON()).toMatchSnapshot();
	});

	it("should be able to call handleNavigate with correct parameters when pressed", () => {
		const { getByTestId } = render(<CartItem data={GAME_MOCK} />);
		const touchableNavigate = getByTestId("cart-item-touchable");

		fireEvent.press(touchableNavigate);

		expect(handleNavigate).toHaveBeenCalledWith({
			id: GAME_MOCK.id,
			router: expect.any(Object),
		});
	});

	it("should be able to call deleteFromCart with correct id when trash icon is pressed", () => {
		const { getByTestId } = render(<CartItem data={GAME_MOCK} />);
		const deleteFromCartButton = getByTestId("cart-item-delete-icon");

		fireEvent.press(deleteFromCartButton);

		const { deleteFromCart } = useCart();

		expect(deleteFromCart).toHaveBeenCalledWith(GAME_MOCK.id);
	});
});
