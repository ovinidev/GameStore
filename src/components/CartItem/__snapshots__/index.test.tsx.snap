// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem should be able to match the snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="cart-item-touchable"
>
  <View
    className="flex flex-row items-center gap-3"
  >
    <Image
      height={65}
      source={
        {
          "uri": "image-url",
        }
      }
      width={65}
    />
    <View
      className="flex gap-2"
    >
      <Text
        className="text-slate-50 text-xl"
      >
        Game Beta
      </Text>
      <Text
        className="self-start font-semibold text-slate-100 text-xl"
      >
        R$ 
        60
      </Text>
    </View>
  </View>
  <Feather
    className="mr-1"
    color="white"
    name="trash"
    onPress={[Function]}
    size={24}
    testID="cart-item-delete-icon"
  />
</View>
`;
